openmls/src/group/mls_group/errors.rs

```diff
@@ -100,3 +92,79 @@ pub enum UnverifiedMessageError {
     #[error(transparent)]
     InvalidCommit(#[from] StageCommitError),
 }
+
+/// Create message error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum CreateMessageError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error("The own CredentialBundle could not be found in the key store.")]
+    NoMatchingCredentialBundle,
+    #[error(transparent)]
+    GroupStateError(#[from] MlsGroupStateError),
+}
+
+/// Add members error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum AddMembersError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error("The own CredentialBundle could not be found in the key store.")]
+    NoMatchingCredentialBundle,
+    #[error(transparent)]
+    EmptyInput(#[from] EmptyInputError),
+    #[error(transparent)]
+    CreateCommitError(#[from] CreateCommitError),
+    #[error(transparent)]
+    GroupStateError(#[from] MlsGroupStateError),
+}
+
+/// Propose add members error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum ProposeAddMemberError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error("The own CredentialBundle could not be found in the key store.")]
+    NoMatchingCredentialBundle,
+    #[error("The new member does not support all required extensions.")]
+    UnsupportedExtensions,
+    #[error(transparent)]
+    GroupStateError(#[from] MlsGroupStateError),
+}
+
+/// Propose remove members error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum ProposeRemoveMemberError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error("The own CredentialBundle could not be found in the key store.")]
+    NoMatchingCredentialBundle,
+    #[error(transparent)]
+    GroupStateError(#[from] MlsGroupStateError),
+}
+
+/// Remove members error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum RemoveMembersError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error("The own CredentialBundle could not be found in the key store.")]
+    NoMatchingCredentialBundle,
+    #[error(transparent)]
+    EmptyInput(#[from] EmptyInputError),
+    #[error(transparent)]
+    CreateCommitError(#[from] CreateCommitError),
+    #[error(transparent)]
+    GroupStateError(#[from] MlsGroupStateError),
+}
+
+/// Leave group error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum LeaveGroupError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error("The own CredentialBundle could not be found in the key store.")]
+    NoMatchingCredentialBundle,
```

Comment:
Duplicating this error in all of these enums makes me think there should be a higher level error that can combine these.

Reply:
This appears as a result of all operations that require signing. Since we try to flatten the errors, I think it makes sense to repeat it in every enum. If it was more than one variant, I agree that we should instead embed a generic error instead.